File: Leetcode 30 Days Solution/Valid Mountain Array.cpp
class Solution {
public:
    bool validMountainArray(vector<int>& arr) {
        
        bool res;
        if(arr.size()<3)
        {
            res=false;
        }
        else 
        {
            int left=0;
            int right=arr.size()-1;
            while(left<right)
            {
                if(arr[left+1]>arr[left])
                {
                    left++;
                }
                else if(arr[right-1]>arr[right])
                {
                    right--;
                }
                else
                {
                    break;
                }
            }
            if(left!=0 && right!=arr.size()-1 && left==right)
            {
                res=true;
            }
            else
            {
                res=false;
            }
        }
        return res;
        
    }
};

File: Leetcode 30 Days Solution/Valid Mountain Array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        
        listlen=len(arr)
        if listlen<3:
            return False
        else:
            leftptr=0
            rightptr=listlen-1
            rightstopindex=-1
            leftstopindex=listlen
            for i in range(rightptr,0,-1):
                if arr[rightptr]<arr[rightptr-1]:
                    rightptr=rightptr-1
                else:
                    rightstopindex=rightptr
                    break
            for i in range(0,listlen-1,1):
                if arr[leftptr+1]>arr[leftptr]:
                    leftptr=leftptr+1
                else:
                    leftstopindex=leftptr
                    break
            if rightstopindex==leftstopindex:
                return True
            else:
                return False
